compiler/qsc/src/interpret/circuit_tests.rs

```diff
@@ -0,0 +1,804 @@
+        q_0    ── H ──── M ──
+                         ╘═══
+    "]]
+    .assert_eq(&circ.to_string());
+}
+
+#[test]
+fn unrestricted_profile_result_comparison() {
+    let mut interpreter = interpreter(
+        r"
+            namespace Test {
+                open Microsoft.Quantum.Measurement;
+                @EntryPoint()
+                operation Main() : Result[] {
+                    use q1 = Qubit();
+                    use q2 = Qubit();
+                    H(q1);
+                    H(q2);
+                    let r1 = M(q1);
+                    let r2 = M(q2);
+                    if (r1 == r2) {
+                        X(q1);
+                    }
+                    ResetAll([q1, q2]);
+                    [r1, r2]
+                }
+            }
+        ",
+        Profile::Unrestricted,
+    );
+
+    interpreter.set_quantum_seed(Some(2));
+
+    let circuit_err = interpreter
+        .circuit(CircuitEntryPoint::EntryPoint)
+        .expect_err("circuit should return error")
+        .pop()
+        .expect("error should exist");
+
+    expect!["Qsc.Eval.ResultComparisonUnsupported"].assert_eq(
+        &circuit_err
+            .code()
+            .expect("error code should exist")
+            .to_string(),
+    );
+
+    let circuit = interpreter.get_circuit();
+    expect![""].assert_eq(&circuit.to_string());
+
+    let mut out = std::io::sink();
+    let mut r = GenericReceiver::new(&mut out);
+
+    // Counterintuitive but expected: result comparisons
+    // are okay if calling get_circuit() after incremental
+    // evaluation, because we're using the current simulator
+    // state.
+    interpreter
+        .eval_fragments(&mut r, "Test.Main();")
+        .expect("eval should succeed");
+
+    let circuit = interpreter.get_circuit();
+    expect![[r"
+        q_0    ── H ──── M ──── X ─── |0〉 ─
+                         ╘═════════════════
+        q_1    ── H ──── M ─── |0〉 ────────
+                         ╘═════════════════
+    "]]
+    .assert_eq(&circuit.to_string());
+}
+
+#[test]
+fn custom_intrinsic() {
+    let mut interpreter = interpreter(
+        r"
+    namespace Test {
+        open Microsoft.Quantum.ResourceEstimation;
+
+        @EntryPoint()
+        operation Main() : Unit {
+            use qs = Qubit[10];
+            AccountForEstimates(
+                [
+                    AuxQubitCount(1),
+                    TCount(2),
+                    RotationCount(3),
+                    RotationDepth(4),
+                    CczCount(5),
+                    MeasurementCount(6),
+                ],
+                PSSPCLayout(),
+                qs);
+        }
+    }",
+        Profile::Unrestricted,
+    );
+
+    let circ = interpreter
+        .circuit(CircuitEntryPoint::EntryPoint)
+        .expect("circuit generation should succeed");
+
+    // This is one gate that spans ten target wires, even though the
+    // text visualization doesn't convey that clearly.
+    expect![[r"
```

Comment:
For this test, I'm not sure how to read the circuit. By looking at the definition of `AccountForEstimatesInternal`, it seems like that operation takes an array of tuples of ints, an integer and a qubit register. However, by looking at the generated circuit, I'm not sure how the arguments match.

Reply:
Yeah, I didn't put a lot effort into custom intrinsics, so the formatting is very unclear here. 

What this supposed to show is all the runtime values of all the classical arguments. Qubit arguments are just shown as wires that go into the gate.

I think this one is extra challenging to read, because the gate spans across 10 qubits and qubit wires aren't even visible (the gate label is too wide for the "column" width I chose). I didn't want to sink too much time into rendering ASCII art nicely, when I expect the SVG image to be the main scenario.

I think I can spend a little more time making the classical arguments look better though. At the moment I'm not grouping tuples or arrays, and a flat list is really hard to read.

Reply:
Thanks for the explanation!

FWIW, I don't think making the classical arguments look better when rendering ASCII is something that needs to be done in this PR. I was just trying to understand the test case, so if you want to make the classical arguments look better in another change, I don't have any problem with that.

Reply:
I ended up changing the formatting a little bit anyway. The classical args (the `[(0,1), (1,2)...` part) get displayed in the SVG circuit as well... so this should help there.

I won't worry about the ASCII art wires for now.